### DISCUSSION TITLE
Disable AutoComplete for file types (Switch to blink.cmp)

### DISCUSSION THREAD

**Discussion by centuryx476** [LazyVim/LazyVim#5123] [General]
Hello, 
With LazyVim 14 and the switch to blink.cmp. Does anyone know how to disable autoComplete for specific file types (.org, .txt) as examples. On the blink.cmp github page it is a little confusing on how to disable autoComplete for specific file types. 

I found this lua function on their README but I cannot seem to get it to work. Has anyone have any luck over here ?
`enabled = function()
  return not vim.tbl_contains({ "lua", "markdown" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
end`

**Comment by dpetka2001** [comment]
I just tried this and opened a lua file and I get no autocompletions. So, it works as expected. You just have to define the filetypes that you don't want to have autocompletions.

**Reply by centuryx476** [reply]
> I just tried this and opened a lua file and I get no autocompletions. So, it works as expected. You just have to define the filetypes that you don't want to have autocompletions.

That's what happens. It works but even if I define the file types it disables for everything.

**Comment by folke** [comment]
That only works on blink main which has not been released yet.
Once a new version is build, you'll also be able to set `vim.b.completion = false` for certain buffers.

**Reply by dpetka2001** [reply]
@folke the `opts.enabled` property works also on 0.7.6. I just tested it with a fresh LazyVim installation from scratch (not my personal config) and I don't get any completions on a Lua file.

**Reply by centuryx476** [reply]
> That only works on blink main which has not been released yet. Once a new version is build, you'll also be able to set `vim.b.completion = false` for certain buffers.

So basically there is no current way to disable for specific file types then ?

**Reply by dpetka2001** [reply]
It seems to work for me. I disabled it for Lua files and then opened a Python file and I get completions in that buffer. Screencast following

[Screencast_2024-12-16-18-25-05.webm](https://github.com/user-attachments/assets/5df2acb4-931c-4f51-85f9-be60ef654dff)

**Reply by centuryx476** [reply]
> It seems to work for me. I disabled it for Lua files and then opened a Python file and I get completions in that buffer. Screencast following
> Screencast_2024-12-16-18-25-05.webm

Can you share the files you edited to get it to work please. 
This is the last thing I need to fix....

**Reply by dpetka2001** [reply]
In the video I showed what i used. It's around the 1:24 mark. Those are the contents that I used for the blink plugin. Nothing more.

**Reply by centuryx476** [reply]
> In the video I showed what i used. It's around the 1:24 mark. Those are the contents that I used for the blink plugin. Nothing more.

Yup, I should have finished the whole video. 
I just tested it and it is disabled for org and text files but in cpp, py files the autoComplete still works.

Thank You very much

**Comment by centuryx476** [comment]
See video in @dpetka2001 comment for solution.

**Reply by aguynamedryan** [reply]
Here's the text from that video:

```lua
return {
  {
    "saghen/blink.cmp",
    opts = {
      enabled = function()
        return not vim.tbl_contains({ "lua", "markdown" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
      end,
    },
  },
}
```
